flexget/plugins/sites/newznab.py

```diff
@@ -1,125 +1,719 @@
+            'force_quotes': {'type': 'boolean', 'default': False},
+            'maxage': {
+                'oneOf': [
+                    {'type': 'string', 'format': 'interval'},
+                    {'type': 'string', 'enum': ['auto']}
+                ]
+            },
+            'use_metadata': {
+                'oneOf': [
+                    {'type': 'boolean'},
+                    one_or_more({'type': 'string', 'enum': list(PLUGIN_LOOKUP_MAP)}, unique_items=True)
+                ]
+            },
         },
-        'required': ['category'],
+        'required': ['api_server_url', 'category'],
         'additionalProperties': False
     }
 
-    def build_config(self, config):
-        log.debug(type(config))
+    def prepare_config(self, config):
+        if 'use_metadata' in config:
+            if 'plugins_list' not in config:
+                if isinstance(config['use_metadata'], basestring):
+                    config['use_metadata'] = [config['use_metadata']]
+                if isinstance(config['use_metadata'], list):
+                    plugins = []
+                    for name in config['use_metadata']:
+                        plugins.append(PLUGIN_LOOKUP_MAP[name])
+                        config['plugins_list'] = plugins
+        if 'category' in config:
+            if 'category_string' not in config:
+                if not isinstance(config['category'], list):
+                    config['category'] = [config['category']]
+                categories = config['category']
+                # Convert named categories to its respective categories id number
+                categories = [c if isinstance(c, int) else CATEGORIES[c] for c in categories]
+                if len(categories) > 0:
+                    config['category_string'] = ','.join(str(c) for c in categories)
+        return config
+
+    def is_tv_entry(self, entry):
+        if safe_get(entry, ['series_name', 'tvdb_id', 'tvrage_id', 'tvmaze_series_id', 'trakt_id'],
+                    [basestring, int]):
+            return True
+        elif safe_get(entry, ['movie_name', 'tmdb_id', 'imdb_id', 'trakt_movie_id'], [basestring, int]):
+            return False
+        else:
+            return None
 
-        if config['category'] == 'tv':
-            config['category'] = 'tvsearch'
+    def has_meta_id(self, entry):
+        return safe_get(entry, ['tvdb_id', 'tvrage_id', 'tvmaze_series_id', 'tmdb_id', 'imdb_id',
+                                'trakt_movie_id', 'trakt_id'], [basestring, int])
 
-        if 'url' not in config:
-            if 'apikey' in config and 'website' in config:
-                params = {
-                    't': config['category'],
-                    'apikey': config['apikey'],
-                    'extended': 1
-                }
-                config['url'] = config['website'] + '/api?' + urlencode(params)
+    def has_supported_meta_id(self, entry):
+        if self.is_tv_entry(entry) is True:
```

Comment:
My function does return multiple value types like most other functions, that use the NoneType as return. In this case it simply signals that the entry is neither a series or "not a" series aka movie. So it will return Nonetype.
As noted this is simply more verbose and reminds me that i need to take special care with If (statement) checks on integer fields, where 0 is a valid value, like `episode_number`.